backend/scripts/sync_prod_to_sqlite.py

```
@@ -0,0 +1,315 @@
+    ])
+    # The password may not appear in argv (visible in ps for the whole dump),
+    # so split it off the userinfo and hand it to libpq via the environment.
+    # Everything else stays untouched: rebuilding the URI wholesale breaks
+    # percent-encoded values and query params such as ?sslmode=require.
+    scheme, _, rest = url.partition('://')
+    userinfo, _, hostpart = rest.rpartition('@')
+    user, _, password = userinfo.partition(':')
+    safe_url = f'{scheme}://{user}@{hostpart}' if userinfo else url
+
+    BACKUP_DIR.mkdir(exist_ok=True)
+    out = BACKUP_DIR / f'tally_prod_{datetime.now():%Y%m%d_%H%M%S}.sql'
```

Ruff (0.16.0)

[warning] `datetime.datetime.now()` called without a `tz` argument

(DTZ005)